plugin/src/test/java/hudson/plugins/swarm/PipelineJobTest.java

```diff
@@ -0,0 +1,132 @@
+package hudson.plugins.swarm;
+
+import static org.junit.Assert.assertNotNull;
+
+import com.google.common.collect.Iterables;
+import hudson.model.Node;
+import hudson.plugins.swarm.test.ProcessDestroyer;
+import hudson.plugins.swarm.test.TestUtils;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.concurrent.atomic.AtomicReference;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.test.steps.SemaphoreStep;
+import org.junit.Assume;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runners.model.Statement;
+import org.jvnet.hudson.test.BuildWatcher;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+import org.jvnet.hudson.test.TemporaryDirectoryAllocator;
+
+public class PipelineJobTest {
+
+    @Rule public RestartableJenkinsRule story = new RestartableJenkinsRule();
+
+    @ClassRule public static BuildWatcher buildWatcher = new BuildWatcher();
+
+    private final ProcessDestroyer processDestroyer = new ProcessDestroyer();
+    private final TemporaryDirectoryAllocator tmpDir = new TemporaryDirectoryAllocator();
```

Comment:
Better to use `org.junit.TemporaryFolder`.

Reply:
Done.